javascripts/discourse/components/kanban/list.gjs

```diff
@@ -0,0 +1,306 @@
+        {{this.renderedTitle}}
+      </div>
+
+      <div class="topics">
+        <ConditionalLoadingSpinner @condition={{this.loading}}>
+          {{#each this.list.topics as |topic|}}
+            <DiscourseKanbanCard
+              @topic={{topic}}
+              @setDragData={{this.setDragData}}
+              @definition={{@definition}}
+            />
+          {{else}}
+            <div class="no_topics">
+              {{i18n (themePrefix "no_topics")}}
+            </div>
+          {{/each}}
+
+          <ConditionalLoadingSpinner @condition={{this.loadingMode}} />
+        </ConditionalLoadingSpinner>
+
+        <div
+          class="list-bottom"
+          {{didInsert this.setupIntersectionObserver}}
+          {{willDestroy this.teardownIntersectionObserver}}
+        ></div>
+      </div>
+    </div>
+  </template>
+
+  @service kanbanHelper;
+  @service dialog;
+  @service modal;
+  @service store;
+
+  @tracked acceptDrag;
+  @tracked loading = false;
+  @tracked loadingMore = false;
+  @tracked list;
+
+  get renderedTitle() {
+    return this.args.definition.title;
+  }
+
+  @action
+  loadTopics() {
+    this.loading = true;
+    this.refreshTopics();
+  }
+
+  refreshTopics() {
+    if (!this.loading && !this.list) {
+      return;
+    }
+    const defaultParams = {
+      match_all_tags: true,
+    };
+
+    const { discoveryTag, discoveryCategory, discoveryParams } =
+      this.kanbanHelper;
+
+    if (discoveryCategory) {
+      defaultParams.category = discoveryCategory.id;
+    }
+
+    const params = Object.assign(
+      {},
+      defaultParams,
+      discoveryParams,
+      this.args.definition.params
+    );
+
+    if (discoveryTag) {
+      params.tags = [...(params.tags || [])];
+      params.tags.push(discoveryTag.id);
+    }
+
+    this.store
+      .findFiltered("topicList", { filter: "latest", params })
+      .then((list) => {
+        this.list = list;
+        this.loading = false;
+      });
+  }
+
+  @action
+  setupIntersectionObserver(element) {
+    this.io = new IntersectionObserver((entries) => {
+      if (entries[0].intersectionRatio <= 0) {
+        return;
+      }
+      this.loadMore();
+    });
+
+    this.io.observe(element);
+  }
+
+  @action
+  teardownIntersectionObserver() {
+    this.io?.disconnect();
+  }
+
+  @action
+  dragLeave(event) {
+    event.preventDefault();
+    this.acceptDrag = false;
+  }
+
+  @action
+  dragOver(event) {
+    event.preventDefault();
+    this.acceptDrag = true;
+  }
+
+  @action
+  drop() {
+    this.dropped();
+    this.acceptDrag = false;
+  }
+
+  @action
+  async loadMore() {
+    if (!this.list && !this.loading) {
+      return this.loadTopics(); // Initial load
+    }
+
+    if (this.loading || !this.list.canLoadMore || this.loadingMore) {
+      return;
+    }
+
+    this.loadingMore = true;
+    await this.list.loadMore();
+    this.loadingMode = false;
+  }
+
+  @action
+  dropped() {
```

Comment:
The complexity of this function is a little bit too much for me

Reply:
Yeah I agree - this is just copy/pasted from the old implementation. I'd prefer not to touch it as part of this PR, but hopefully it's something we can improve in the not-too-distant future.